### PR TITLE
[BugFix][FishSpeech] Fix structured voice clone prefill conditioning

### PR DESCRIPTION
**Pull request by Sy0307**
## Summary

This patch fixes the Fish Speech S2 Pro voice clone regression reported in #2394.

The root cause was that the structured voice clone prefill path only rebuilt the prompt from semantic token IDs, but did not add the full DAC codebook embeddings back onto the reference-audio span. Decode-time `talker_mtp()` still used full codebook conditioning, so prefill and decode saw different conditioning signals and the cloned voice drifted or collapsed.

This patch:

- adds `encode_reference_audio_codes()` to obtain full DAC codes from the reference audio
- updates the structured voice clone prefill path to rebuild semantic prompt IDs from codebook 0
- injects full DAC codebook embeddings back into the `<|audio_start|> ... <|audio_end|>` span
- applies the same `/ sqrt(num_codebooks + 1)` scaling used by decode-time conditioning
- fixes a mixed-batch indexing bug in the Fish Speech DAC decoder when empty requests are present

## Verification

- `.venv/bin/ruff format --check ...` on the 4 changed Python files
- `python -m pytest -q tests/model_executor/models/test_fish_speech_regressions.py` -> `2 passed`
- local streaming benchmark:
  - current patch: `TTFB 371.2ms`, `TTFP 803.8ms`, `total 4791.2ms`
  - clean baseline: `TTFB 370.8ms`, `TTFP 806.8ms`, `total 4470.3ms`
- local voice clone sample check:
  - reference: `RMS 0.02379`, `peak 0.1961`
  - broken baseline: `RMS 0.00393`, `peak 0.0544`
  - current patch: `RMS 0.02302`, `peak 0.1902`

## Notes

cc @linyueqian

### CODE REVIEW

File: tests/model_executor/models/test_fish_speech_regressions.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        audio_lengths = torch.tensor([100], dtype=torch.long)
+        return wav, audio_lengths
+
+
+class _FakeFastAR:
+    def __init__(self, codes: torch.Tensor):
+        self._codes = codes
+
+    def __call__(self, **kwargs):
+        del kwargs
```
Comment (lishunyang12) [inline tests/model_executor/models/test_fish_speech_regressions.py]: `_FakeFastAR` is defined but unused — leftover from a draft?

File: vllm_omni/model_executor/models/fish_speech/dac_encoder.py [hunk 2]
```diff
@@ -66,42 +66,21 @@ def _load_dac_codec(
 def _get_resample_kernel(
     source_sr: int,
     target_sr: int,
-    device_type: str,
-    device_index: int | None,
-    dtype_name: str,
+    device: torch.device,
+    dtype: torch.dtype,
 ):
     import torchaudio
```
Comment (lishunyang12) [inline vllm_omni/model_executor/models/fish_speech/dac_encoder.py:72]: The original signature used primitive types (`str`, `int | None`) specifically because this function is decorated with `@lru_cache(maxsize=16)`. `torch.device` and `torch.dtype` happen to be hashable so it still works, but it's a subtle invariant — worth a brief comment so the next person doesn't swap in an unhashable type.

File: vllm_omni/model_executor/models/fish_speech/fish_speech_slow_ar.py [hunk 3]
```diff
@@ -547,7 +548,34 @@ def _build_structured_voice_clone_prefill_embeds(self, info_dict: dict[str, Any]
...
+
+        frames_in_prompt = e - s
+        if ref_codes_fq.device != embeds.device:
+            ref_codes_fq = ref_codes_fq.to(device=embeds.device, dtype=torch.long)
+        frames = min(int(ref_codes_fq.shape[0]), int(frames_in_prompt))
+        if frames <= 0:
+            return embeds
+
+        q = min(int(ref_codes_fq.shape[1]), self._num_codebooks)
+        offsets = (torch.arange(q, device=embeds.device, dtype=torch.long) * self._codebook_size).unsqueeze(0)
```
Comment (lishunyang12) [inline vllm_omni/model_executor/models/fish_speech/fish_speech_slow_ar.py:572]: Silent `clamp(min=0)` could mask corrupted codec output. Any reason not to log a warning when negative codes appear?

**Comment by mru4913** [comment]
All good. It can be merged. @Sy0307 👍 good job！

**Comment by Sy0307** [comment]
> All good. It can be merged. @Sy0307 👍 good job！

Can you close #2394 as completed? Thanks. Feel free to give more feedback now.
